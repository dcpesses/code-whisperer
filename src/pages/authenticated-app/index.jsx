/* eslint-disable react/no-unused-state */
/* eslint-disable no-console */
import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import LoadingRipple from '@/components/loading-ripple';
import {Navigate} from 'react-router-dom';
import Login from '@/pages/login';
import TwitchApi from '@/api/twitch';
import {withRouter, Debounce} from '@/utils';
import MainScreen from '@/components/main-screen';
import { clearUserInfo, setModeratedChannels, setUserInfo } from '@/features/player-queue/user-slice.js';
import { clearChannelInfo, clearModerators, clearVIPs, setChannelInfo, setModerators, setVIPs } from '@/features/twitch/channel-slice.js';

export const noop = () => void 0;

const TWITCH_API = new TwitchApi({
  redirectUri: import.meta.env.VITE_APP_REDIRECT_URI_NOENCODE,
  clientId: import.meta.env.VITE_APP_TWITCH_CLIENT_ID,
  clientSecret: import.meta.env.VITE_APP_TWITCH_CLIENT_SECRET,
  debug: true,
  init: false
});

class AuthenticatedApp extends Component {
  static get propTypes() {
    return {
      clearChannelInfo: PropTypes.func,
      clearModerators: PropTypes.func,
      clearUserInfo: PropTypes.func,
      clearVIPs: PropTypes.func,
      setChannelInfo: PropTypes.func,
      setModeratedChannels: PropTypes.func,
      setModerators: PropTypes.func,
      setUserInfo: PropTypes.func,
      setVIPs: PropTypes.func,
    };
  }
  static get defaultProps() {
    return {
      clearChannelInfo: noop,
      clearModerators: noop,
      clearUserInfo: noop,
      clearVIPs: noop,
      setChannelInfo: noop,
      setModeratedChannels: noop,
      setModerators: noop,
      setUserInfo: noop,
      setVIPs: noop,
    };
  }
  constructor(props) {
    super(props);
    this.state = {
      userInfo: {},
      username: localStorage.getItem('__username') || '',
      user_id: localStorage.getItem('__user_id') || 0,
      profile_image_url: localStorage.getItem('__profile_image_url') || '',
      accessToken: localStorage.getItem('__access_token') || '',
      auth_pending: false,
      failed_login: false,
      has_logged_out: false,
      moderators: null,
      vips: null,
    };

    this.twitchApi = TWITCH_API;

    this.hasAlreadyInit = false;
    this._isMounted = false;

    this.twitchAuthReady = false;

    this.onDelayedMount = Debounce(this.onMount.bind(this), 50);

    this.showLoginButton = true;
  }

  componentDidMount() {
    this._isMounted = true;
    console.log('authenticated-app - componentDidMount');
    this.onDelayedMount();
  }
  componentWillUnmount() {
    this._isMounted = false;
    this.props.clearChannelInfo();
    this.props.clearModerators();
    this.props.clearUserInfo();
    this.props.clearVIPs();
    this.twitchApi.closeChatClient();
    console.log('authenticated-app - componentWillUnmount');
  }

  promisedSetState = (newState) => new Promise(resolve => this.setState(newState, resolve));

  onMount = async() => {
    if (this._isMounted !== true) {
      console.log('authenticated-app - onDelayedMount: not mounted');
    }
    let accessToken = localStorage.getItem('__access_token');
    let refreshToken = localStorage.getItem('__refresh_token');
    let expiryTime = parseInt(localStorage.getItem('__expiry_time'), 10);
    let currTime = Date.now();
    if (accessToken && accessToken !== 'undefined' && !isNaN(expiryTime)) {
      if (expiryTime > currTime) {
        // try using existing access token
        console.log('onDelayedMount - trying previous token', {
          localStorage: accessToken
        });
        try {
          this.twitchApi.accessToken = accessToken;
          const resumeResponse = await this.twitchApi.resume();
          if (resumeResponse && !resumeResponse.error) {
            console.log('onDelayedMount - successful resume response!', resumeResponse);
            return this.onTwitchAuthInit();
          }
          console.log('onDelayedMount - unexpected resume response', resumeResponse);
        } catch (e) {
          console.log('onDelayedMount - previous init not valid', e);
        }
      } else if (refreshToken) {
        // try using existing refresh token
        console.log('onDelayedMount - trying refresh token', {
          localStorage: refreshToken
        });
        try {
          this.twitchApi.refreshToken = refreshToken;
          const refreshResponse = await this.twitchApi.requestRefreshToken();
          if (refreshResponse && !refreshResponse.error) {
            console.log('onDelayedMount - successful requestRefreshToken response!', refreshResponse);
          }
          const resumeResponse = await this.twitchApi.resume(refreshResponse.access_token, refreshResponse);
          if (resumeResponse && !resumeResponse.error) {
            console.log('onDelayedMount - successful requestRefreshToken response!', resumeResponse);
            return this.onTwitchAuthInit();
          }
          console.log('onDelayedMount - unexpected requestRefreshToken response', resumeResponse);
        } catch (e) {
          console.log('onDelayedMount - previous init not valid', e);
        }
      }
    }
    console.log('onDelayedMount - calling init');
    try {
      const initResponse = await this.twitchApi.init();
      console.log('onDelayedMount - initialized response:', initResponse);
      if (!initResponse) {
        throw 'No response';
      }
      if (initResponse.error) {
        throw initResponse.error;
      }
      return this.onTwitchAuthInit();
    } catch (e) {
      console.log('onDelayedMount - error initializing:', e);
      return this.onTwitchAuthError();
    }
  };

  onTwitchAuthInit = () => {
    let userInfo = this.twitchApi.userInfo;
    console.log('onTwitchAuthInit', {userInfo});
    if (!userInfo?.login) {
      console.log('onTwitchAuthInit - no user info');
      return this.setState({
        auth_pending: false,
        failed_login: true,
      });
    }
    this.props.setUserInfo(userInfo);
    this.props.setChannelInfo(userInfo);
    this.setState({
      username: userInfo.login,
      user_id: userInfo.id,
      profile_image_url: userInfo.profile_image_url,
      auth_pending: false,
      failed_login: false,
    }, this.updateModsAndVIPs);
  };

  onTwitchAuthError = () => {
    console.log('authenticated-app - onTwitchAuthError');
    return this.setState({
      auth_pending: false,
      failed_login: true,
    });
  };

  logOut = async() => {
    console.log('logOut');

    try {
      await this.twitchApi.logOut();
      localStorage.removeItem('__error_msg');
      console.log('logOut: setState - has_logged_out');
      return this.setState({
        has_logged_out: true
      });
    } catch (e) {
      console.log('logOut: error', e);
      return this.setState({
        has_logged_out: true
      });
    }
  };

  updateModeratedChannels = async() => {
    try {
      await this.twitchApi.validateToken();
      const id = this.state.user_id;
      let moderatedChannels = await this.twitchApi.requestModeratedChannels(id);
      if (moderatedChannels.data) {
        this.props.setModeratedChannels(moderatedChannels.data);
      }
    } catch (e) {
      console.warn('updateModeratedChannels: error', e);
    }
  };

  updateModsAndVIPs = async() => {
    try {
      await this.twitchApi.validateToken();
      this.updateModerators();
      this.updateVIPs();
      await this.updateModeratedChannels();
    } catch (e) {
      console.warn('updateModsAndVIPs: error', e);
    }
  };

  updateModerators = async() => {
    const props = this.props;
    try {
      const id = this.state.user_id;
      const moderators = await this.twitchApi.requestModerators(id);
      props.setModerators(moderators);
    } catch (e) {
      console.warn('updateModerators: error', e);
    }
  };

  updateVIPs = async() => {
    try {
      const id = this.state.user_id;
      const vips = await this.twitchApi.requestVIPs(id);
      this.props.setVIPs(vips);
    } catch (e) {
      console.warn('updateVIPs: error', e);
    }
  };

  render = () => {
    if (this._isMounted && (this.state.failed_login === true || this.state.has_logged_out === true)) {
      if (this.showLoginButton) {
        return (<Login />);
      }
      console.log('render: navigate to login');
      return (<Navigate to="/login" />);
    }

    let mainContent = (
      <div className="col display-6 text-center full-pg">
        <div></div>
        <div>
          <LoadingRipple />
          <br />
          Loading...
        </div>
        <div></div>
      </div>
    );

    if (this.state.username && this.twitchApi?.isChatConnected) {
      mainContent = (
        <MainScreen
          access_token={this.twitchApi?.accessToken}
          channel={this.state.username}
          moderators={this.state.moderators}
          onLogOut={this.logOut}
          twitchApi={this.twitchApi}
          vips={this.state.vips}
        />
      );
    }

    return (
      <div id="authenticated-app container text-center">
        {mainContent}
      </div>
    );
  };
}

export {AuthenticatedApp};

const mapDispatchToProps = () => ({
  clearChannelInfo,
  clearModerators,
  clearUserInfo,
  clearVIPs,
  setChannelInfo,
  setModeratedChannels,
  setModerators,
  setUserInfo,
  setVIPs,
});
export default connect(
  null,
  mapDispatchToProps()
)(withRouter(AuthenticatedApp));
