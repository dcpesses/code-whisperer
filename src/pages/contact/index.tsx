import { useState } from 'react';
import { Link } from 'react-router-dom';
import ContactForm from '@/components/contact-form';

function Contact() {

  const [validated, setValidated] = useState(false);

  const handleSubmit = (event: SubmitEvent) => {
    const form:HTMLFormElement = event.currentTarget as HTMLFormElement;
    if (form.checkValidity() === false) {
      event.preventDefault();
      event.stopPropagation();
    }
    setValidated(true);
  };

  const formProps = {
    action: '/thanks',
    acceptCharset: 'UTF-8',
    encType: 'multipart/form-data',
    method: 'GET',
  };
  return (
    <div className="contact container" data-testid="contact">
      <main className="px-3">
        <h1 className="py-4">Contact Us</h1>
        <ContactForm
          disabled={false}
          formProps={formProps}
          onSubmit={handleSubmit}
          validated={validated}
        />
        <div className="col-6 mx-auto">
          <Link className="btn btn-outline-primary btn-lg mt-5" to="/">
            Go Home
          </Link>
        </div>
      </main>
    </div>
  );
}

export default Contact;
