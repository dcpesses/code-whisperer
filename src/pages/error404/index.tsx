import { Link } from 'react-router-dom';

function Error404() {
  return (
    <div className="error-404 full-pg d-flex w-100 h-100 my-0 mx-auto flex-column">
      <main className="px-3 text-center">
        <h1 className="display-1 ppb-4">404</h1>
        <p>
          <small className="fst-italic">
            I&apos;m sorry, Dave. I&apos;m afraid I can&apos;t do that.
          </small>
        </p>
        <div className="col-6 mx-auto">
          <Link className="btn btn-outline-primary btn-lg mt-5" to="/">
            Go Home
          </Link>
        </div>
      </main>
    </div>
  );
}

export default Error404;
