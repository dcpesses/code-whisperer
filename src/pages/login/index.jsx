import React, {Component} from 'react';
import {version} from '../../../package.json';

import logo from '@/assets/new-logo.svg';
import './login.css';

const scopes = 'chat:read chat:edit moderation:read user:manage:whispers';
const loginUrl = 'https://id.twitch.tv/oauth2/authorize'
    + `?client_id=${import.meta.env.VITE_APP_TWITCH_CLIENT_ID}`
    + `&response_type=code&scope=${scopes}`
    + `&redirect_uri=${import.meta.env.VITE_APP_REDIRECT_URI}`;

class Login extends Component {

  render() {
    return (
      <div id="login-screen" className="container full-pg fade-in">
        <div className="text-center pb-4">
          <img src={logo} className="login-logo w-50 mx-auto mb-4" alt="Logo" />
        </div>

        <a href={loginUrl} className="btn btn-sm fs-2 py-2 px-3 rounded-4">
          Log In With <strong>Twitch</strong>
        </a>

        <div>
          <small>
            {`v${version}`}
          </small>
        </div>
      </div>
    );
  }
}

export default Login;
