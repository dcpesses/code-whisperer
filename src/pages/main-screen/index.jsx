import {useState} from 'react';
import PropTypes from 'prop-types';
import Header from '@/components/header';
import GameCodeForm from '@/components/game-code-form';
import GameQueue from '@/components/game-queue';
import TestUserForm from '@/components/test-user-form';

import './main-screen.css';

export const noop = ()=>{};

function MainScreen({
  accessToken, onLogOut, profile_image_url, toggleDeprecatedView, user_id, username, updateUsername
}) {

  let [roomCode, setRoomCode] = useState('');

  return (
    <div className="main-screen">
      <Header
        accessToken={accessToken}
        toggleDeprecatedView={toggleDeprecatedView}
        onLogOut={onLogOut}
        profile_image_url={profile_image_url}
        user_id={user_id}
        username={username}
      />

      <div id="content" className="container mx-auto">

        <GameCodeForm
          roomCode={roomCode}
          sendCodeToAll={window.console.log}
          setRoomCode={setRoomCode}
        />

        <div className="queues d-flex flex-column flex-md-row my-2">
          <GameQueue label="play queue" />
          <GameQueue label="interested" />
        </div>

        <TestUserForm
          user={{username}}
          updateUsername={updateUsername}
          onLogOut={onLogOut}
        />

      </div>
    </div>

  );
}

MainScreen.propTypes = {
  accessToken: PropTypes.any,
  onLogOut: PropTypes.func,
  profile_image_url: PropTypes.any,
  toggleDeprecatedView: PropTypes.func,
  updateUsername: PropTypes.func,
  user_id: PropTypes.any,
  username: PropTypes.any,
};

MainScreen.defaultProps = {
  accessToken: '',
  onLogOut: noop,
  profile_image_url: '',
  toggleDeprecatedView: noop,
  updateUsername: noop,
  user_id: '',
  username: '',
};
export default MainScreen;
