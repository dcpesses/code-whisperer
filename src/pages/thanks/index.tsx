import { Link } from 'react-router-dom';

function Thanks() {
  return (
    <div className="thanks container-fluid d-flex justify-content-center align-items-center vw-100 vh-100">
      <main className="row text-center d-flex align-items-center">
        <h1 className="pb-4">Thanks!</h1>
        <p>
          We&apos;ve received your submission.
        </p>
        <div className="col-6 mx-auto">
          <Link className="btn btn-outline-primary mt-5" to="/">
            Back
          </Link>
        </div>
      </main>
    </div>
  );
}

export default Thanks;
