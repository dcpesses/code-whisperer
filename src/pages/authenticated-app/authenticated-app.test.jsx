/* eslint-disable testing-library/render-result-naming-convention */
/* eslint-env jest */
import {AuthenticatedApp} from '@/pages/authenticated-app';
import {createRenderer} from 'react-test-renderer/shallow';
// import MainScreen from '../landing/MainScreen';
import Login from '@/pages/login';
import React from 'react';
import {vi} from 'vitest';

// vi.mock('../landing/MainScreen');
vi.mock('react-router-dom', () => {
  const reactRouterDom = vi.importActual('react-router-dom');
  return {
    ...reactRouterDom,
    // eslint-disable-next-line react/display-name
    Navigate: () => () => (<div>React Tooltip Mock</div>),
    redirect: vi.fn()
  };
});

describe('AuthenticatedApp', () => {

  let props = {
    history: {
      length: 3,
      action: 'POP',
      location: {
        pathname: '/',
        search: '?code=foobar&scope=chat:read chat:edit moderation:read whispers:edit',
        hash: ''
      }
    },
    match: {
      path: '/',
      url: '/',
      isExact: true,
      params: {}
    }
  };

  describe('componentDidMount', () => {
    test('should call onDelayedMount', () => {
      let component = new AuthenticatedApp();
      vi.spyOn(component, 'onDelayedMount').mockResolvedValue(null);

      component.componentDidMount();
      expect(component.onDelayedMount).toHaveBeenCalledTimes(1);
    });
  });

  describe('onMount', () => {

    beforeEach(() => {
      vi.useFakeTimers();
      // "Calm down, Marty, Einstein and the car are completely intact."
      vi.spyOn(Date, 'now').mockReturnValue(499162860000); // October 26, 1985 1:21:00 AM PST
    });
    afterEach(() => {
      vi.useRealTimers();
    });
    test('should reuse existing access token from localStorage if available', async() => {
      vi.spyOn(window.localStorage.__proto__, 'getItem').mockImplementation((label) => {
        if (label === '__access_token') {
          return 'MOCK TOKEN';
        }
        if (label === '__expiry_time') {
          return '499163700000'; // October 26, 1985 1:35:00 AM PST
        }
        return null;
      });
      let component = new AuthenticatedApp();
      vi.spyOn(component, 'onTwitchAuthInit');
      component.twitchApi = {
        accessToken: null,
        resume: vi.fn().mockResolvedValue({
          oauth: {},
          users: {},
          valid: {},
        })
      };

      await component.onMount();
      expect(component.twitchApi.accessToken).toBe('MOCK TOKEN');
      expect(component.onTwitchAuthInit).toHaveBeenCalled();
    });

    test('should initialize the Twitch API class and call onTwitchAuthInit when completed', async() => {
      vi.spyOn(window.localStorage.__proto__, 'getItem').mockReturnValue(undefined);
      let component = new AuthenticatedApp();
      vi.spyOn(component, 'onTwitchAuthInit');
      component.twitchApi = {
        _accessToken: null,
        init: vi.fn().mockResolvedValue({auth: true, users: true})
      };

      await component.onMount();
      expect(component.onTwitchAuthInit).toHaveBeenCalled();
    });
  });

  describe('onTwitchAuthInit', () => {
    test('should handle response with user info', () => {
      const component = new AuthenticatedApp();
      vi.spyOn(component, 'setState');
      const props = {
        setChannelInfo: vi.fn(),
        setUserInfo: vi.fn(),
      };
      component.props = props;
      component.twitchApi = {
        userInfo: {
          login: 'mockUsername',
          id: 'mockUserId',
          profile_image_url: 'mockProfileImageUrl',
        }
      };

      component.onTwitchAuthInit();
      expect(props.setUserInfo).toHaveBeenCalledTimes(1);
      expect(props.setChannelInfo).toHaveBeenCalledTimes(1);
      expect(component.setState).toHaveBeenCalledTimes(1);
      expect(component.setState).toHaveBeenCalledWith({
        username: 'mockUsername',
        user_id: 'mockUserId',
        profile_image_url: 'mockProfileImageUrl',
        auth_pending: false,
        failed_login: false,
      }, component.updateModsAndVIPs);
    });

    test('should handle response with no user info', () => {
      let component = new AuthenticatedApp();
      component.twitchApi._userInfo = {};
      vi.spyOn(component, 'setState');

      component.onTwitchAuthInit();

      expect(component.setState).toHaveBeenCalledWith({
        auth_pending: false,
        failed_login: true,
      });
    });
  });

  describe('onTwitchAuthError', () => {
    test('should set state with failed login', () => {
      let component = new AuthenticatedApp();
      vi.spyOn(component, 'setState');

      component.onTwitchAuthError();
      expect(component.setState).toHaveBeenCalledWith({
        auth_pending: false,
        failed_login: true,
      });
    });
  });

  describe('logOut', () => {
    const {location} = window;

    beforeAll(() => {
      delete window.location;
      window.location = {reload: vi.fn()};
    });

    afterAll(() => {
      window.location = location;
    });
    test('should log out of api and update has_logged_out state', async() => {
      vi.spyOn(window.localStorage.__proto__, 'removeItem');
      let component = new AuthenticatedApp();
      vi.spyOn(component, 'setState');
      component.twitchApi = {
        logOut: vi.fn().mockResolvedValue({})
      };
      await component.logOut();

      expect(window.localStorage.__proto__.removeItem).toHaveBeenCalled();
      expect(component.setState).toHaveBeenCalled();
    });
    test('should handle error and update has_logged_out state', async() => {
      vi.spyOn(window.localStorage.__proto__, 'removeItem');
      let component = new AuthenticatedApp();
      vi.spyOn(component, 'setState');
      component.twitchApi = {
        logOut: vi.fn().mockRejectedValue({})
      };
      await component.logOut();

      expect(component.setState).toHaveBeenCalled();
      expect(window.localStorage.__proto__.removeItem).not.toHaveBeenCalled();
    });
  });

  describe('render', () => {
    test('should render with MainScreen', () => {
      const shallowRenderer = createRenderer();
      shallowRenderer.render(<AuthenticatedApp {...props} />);
      let instance = shallowRenderer.getMountedInstance();
      instance.twitchApi = {mock: 'TwitchApi'};
      instance.setState({
        access_token: 'yadayadayada',
        failed_login: false,
        moderators: [],
        username: 'sirgoosewell'
      });
      let component = shallowRenderer.getRenderOutput();
      // expect(component.props.children.type).toBe(MainScreen);
      expect(component).toMatchSnapshot();
      shallowRenderer.unmount();
    });
    test('should render with Login on failed login', () => {
      const shallowRenderer = createRenderer();
      shallowRenderer.render(<AuthenticatedApp {...props} />);
      let instance = shallowRenderer.getMountedInstance();
      instance._isMounted = true;
      instance.setState({
        access_token: null,
        failed_login: true,
        username: null
      });
      let component = shallowRenderer.getRenderOutput();
      expect(component.type).toBe(Login);
      expect(component).toMatchSnapshot();
      shallowRenderer.unmount();
    });
    test('should render with Login on has_logged_out state', () => {
      const shallowRenderer = createRenderer();
      shallowRenderer.render(<AuthenticatedApp {...props} />);
      let instance = shallowRenderer.getMountedInstance();
      instance._isMounted = true;
      instance.setState({
        access_token: null,
        failed_login: false,
        has_logged_out: true,
        username: null
      });
      let component = shallowRenderer.getRenderOutput();
      expect(component.type).toBe(Login);
      expect(component).toMatchSnapshot();
      shallowRenderer.unmount();
    });
    test('should render with null', () => {
      const shallowRenderer = createRenderer();
      shallowRenderer.render(<AuthenticatedApp {...props} />);
      let instance = shallowRenderer.getMountedInstance();
      instance.setState({
        access_token: null,
        failed_login: false,
        username: null
      });
      let component = shallowRenderer.getRenderOutput();
      expect(component.props.children).toBeDefined();
      expect(component).toMatchSnapshot();
      shallowRenderer.unmount();
    });
  });
});
