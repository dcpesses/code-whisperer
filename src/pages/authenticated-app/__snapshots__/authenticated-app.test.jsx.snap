// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AuthenticatedApp > render > should render with Login on failed login 1`] = `<Login />`;

exports[`AuthenticatedApp > render > should render with Login on has_logged_out state 1`] = `<Login />`;

exports[`AuthenticatedApp > render > should render with MainScreen 1`] = `
<div
  id="authenticated-app container text-center"
>
  <ImportedMainScreen
    channel="sirgoosewell"
    modList={[]}
    onLogOut={[Function]}
    profile_image_url=""
    toggleDeprecatedView={[Function]}
    twitchApi={
      {
        "mock": "TwitchApi",
      }
    }
    updateUsername={[Function]}
    userInfo={{}}
    user_id={0}
    username="sirgoosewell"
  />
</div>
`;

exports[`AuthenticatedApp > render > should render with null 1`] = `
<div
  id="authenticated-app container text-center"
>
  <div
    className="col display-6 text-center full-pg"
  >
    <div />
    <div>
      <LoadingRipple />
      <br />
      Loading...
    </div>
    <div />
  </div>
</div>
`;
